StringUtil.java

```diff
@@ -7,19 +6,71 @@
  */
 public class StringUtil {
 
-    private static String[] URLParser(String url){
+	private static String[] strSplitArr = null;
+	private static String strReverse ="";
+   private static String[] URLParser(String url){
         //WRITE YOUR CODE HERE..
-        //parse the url and get all the get parameters and return an array of Strings
+	    //parse the url and get all the get parameters and return an array of Strings
         //Input: google.com/api?param1=value1&param2=value2
         //Output: array of strings [] = { {param1:value1}, {param2:value2} }
+	   int index = url.indexOf("?");
+	   String s= url.substring(index+1);
+	   strSplitArr = s.split("&"); 
+	   for(int i=0 ; i< strSplitArr.length ; i++){
+		   strSplitArr[i] ="{" +  strSplitArr[i].replaceAll("=",":")+"}";
+		  // System.out.println("param=" + strSplitArr[i] );
+	   }
+	   return strSplitArr;
+	   
 
+	   
     }
     private static String reverseWords(String sentence){
         //WRITE YOUR CODE HERE..
+    	if(sentence != null){
+    	strSplitArr = sentence.split(" "); //splitting the string 
+    	//System.out.println("strSplitLenght=" + strSplitArr.length);
+    	for (int i= strSplitArr.length-1; i >= 0; i--){
+    		if(i>0){
+    		strReverse += strSplitArr[i] + " ";
+    		}
+    		else
+    		{
+    			strReverse += strSplitArr[i];
+    		}
+    		
+    	}
+    	
+    	//System.out.println("strReverse=" + strReverse);
+    	return strReverse;
+    	}
+    	else{
+    		return null;
+    	}
     }
 
-    private static String reverseString(String sentence){
+  private static String reverseString(String sentence){
         //WRITE YOUR CODE HERE..
+	  if(sentence !=null){
+		  String strReverse = "";
+		  strSplitArr = sentence.split(" ");
+		  //System.out.println("strSplitLenght=" + strSplitArr.length);
+		  for (int i = 0 ;i < strSplitArr.length; i++){
+			  	for ( int j =strSplitArr[i].length()-1 ; j>=0 ; j-- ){
```

Comment:
For better readability the inner for loop can be moved to a method reverseWord() 👍